### PR TITLE
IPv6 hostname in URL support

### LINKED ISSUE #5759: URL parser does not handle IPv6 URLs
Currently the URL parser can handle hostnames, IP addresses, etc.
```
> new URL('http://127.0.0.1')
URL { href: "http://127.0.0.1/", origin: "http://127.0.0.1", protocol: "http:", username: "", password: "", host: "127.0.0.1", hostname: "127.0.0.1", port: "", pathname: "/", hash: "", search: "" }
```

Unfortunately, when using IPv6 addresses in URLs, the hostname is parsed as ```[```, which means the remaining part is treated as a port (and therefore invalid)
```
> new URL('http://[::1]')
Uncaught TypeError: Invalid URL.
    at new URLImpl ($deno$/web/url.ts:418:13)
    at <unknown>:1:1
    at evaluate ($deno$/repl.ts:45:34)
    at Object.replLoop ($deno$/repl.ts:136:13)
```

This has also caused an (uncaught) error in Oak when a URL is accessed by IPv6 address directly, and the same may be true for other applications.

### PR DESCRIPTION
**Pull request by realJoshByrnes**
Fixes #5759 

Handles IPv6 URLs, such as `http://[::1]/`
Should not change the parsing of any other type of URL.

### CODE REVIEW

**Comment by marcosc90** [comment]
@realJoshByrnes The lint check is failing, you can run: `./tools/lint.py` & `./tools/format.py` to fix the issues.

File: cli/tests/unit/url_test.ts [hunk 1]
```diff
@@ -29,6 +29,28 @@ unitTest(function urlParsing(): void {
...
+  assertEquals(urlv6.search, "?foo=bar&baz=12");
+  assertEquals(urlv6.searchParams.getAll("foo"), ["bar"]);
+  assertEquals(urlv6.searchParams.getAll("baz"), ["12"]);
+  assertEquals(urlv6.username, "foo");
+  assertEquals(
+    String(urlv6),
+    "https://foo:bar@[::1]:8000/qux/quux?foo=bar&baz=12#qat"
+  );
+  assertEquals(
+    JSON.stringify({ key: urlv6 }),
```
Comment (bartlomieju) [inline cli/tests/unit/url_test.ts:51]: This test is still failing. @realJoshByrnes can you take a look? I'd like to land this PR for v1.1

**Review by bartlomieju — DISMISSED** [review]

**Comment by realJoshByrnes** [comment]
I'm on it.

**Review by ry — APPROVED** [review]
LGTM - thanks @realJoshByrnes

**Pull request merged**